concept/thing/Relation.java

```diff
@@ -36,12 +39,18 @@
 
     void removePlayer(RoleType roleType, Thing player);
 
-    FunctionalIterator<? extends Thing> getPlayers(String roleType, String... roleTypes);
+    FunctionalIterator<Thing> getPlayers();
 
-    FunctionalIterator<? extends Thing> getPlayers(RoleType... roleTypes);
+    FunctionalIterator<Thing> getPlayers(String[] roleTypes);
+
+    FunctionalIterator<Thing> getPlayers(RoleType[] roleTypes);
+
+    Forwardable<Thing, Order.Asc> getPlayers(String roleType, String... roleTypes);
+
+    Forwardable<Thing, Order.Asc> getPlayers(RoleType roleType, RoleType... roleTypes);
```

Comment:
we make the `Relation` API more expressive, splitting `getPlayers` into variants that return `forwardable` and regular iterators.

Ideally, we would have two variants that look like this:
```
Functionaliterator<> getPlayers(String roleType...)
```
and
```
Forwardable<> getPlayers(String roleType, String... roleTypes);
```

which effectively switch the return type when using 1 or more role types to a forwardable one. However, this is ambiguous and not allowed by java.

To allow passing a variable number of role types instead, we use an `String[]`, which is as useful as `String...` (they are interchangable, but the caller has to construct an array now).